src/Microsoft.AspNetCore.SignalR.Common/Internal/Encoders/Base64Encoder.cs

```diff
@@ -23,18 +23,29 @@ public ReadOnlySpan<byte> Decode(byte[] payload)
             return decoded.Slice(0, written);
         }
 
+        private const int Int32OverflowLength = 10;
+
         public byte[] Encode(byte[] payload)
         {
-            Span<byte> buffer = new byte[Base64.GetMaxEncodedToUtf8Length(payload.Length)];
+            //The format is: [{length}:{message};] so allocate enough to be able to write the entire message
+            Span<byte> buffer = new byte[Int32OverflowLength + 1 + Base64.GetMaxEncodedToUtf8Length(payload.Length) + 1];
```

Comment:
**Question:** Could this result in integer overflow if `Base64.GetMaxEncodedToUtf8Length(payload.Length)` is close to int.MaxValue? Is the expectation that the payload is relatively small?

Reply:
We expect the messages to be relatively small but in theory we support messages up to 2GB before encoding. In this particular case - when a message needs to be base64 encoded - even a smaller original message may not be able to be encoded. I guess I would just check if the message could potentially exceed 2GB and throw.

Reply:
> We expect the messages to be relatively small but in theory we support messages up to 2GB before encoding. 

GetMaxEncodedToUtf8Length() already throws for any payload.Length larger than 1610612733.
https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/Buffers/Text/Base64Encoder.cs#L107

Would that be a concern?

Reply:
I know, I checked. I think it is fine. Any exception from this code path closes the connection and the stack trace should make it clear what happened. I don't expect anyone hitting this exception though - the biggest message someone tried to send was a few MBs and to hit this code path the user would have to use binary formats (e.g msgpack) over text transport (serverSentEvents) neither of which is the default.